reference-implementation/test/readable-byte-stream.js

```diff
@@ -0,0 +1,1415 @@
+    t.equals(result.done, false, 'done');
+
+    const view = result.value;
+    t.equals(view.constructor, Uint8Array, 'value.constructor');
+    t.equals(view.buffer.byteLength, 16, 'value.buffer.byteLength');
+    t.equals(view.byteOffset, 8, 'value.byteOffset');
+    t.equals(view.byteLength, 8, 'value.byteLength');
+    t.equals(view[0], 0x02);
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: getReader(), enqueue(), close(), then read()', t => {
+  let controller;
+
+  const rbs = new ReadableByteStream({
+    start(c) {
+      controller = c;
+    },
+    pull() {
+      t.fail('pull must not be called');
+      t.end();
+    },
+    pullInto() {
+      t.fail('pullInto must not be called');
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  controller.enqueue(new Uint8Array(16));
+  controller.close();
+
+  reader.read().then(result => {
+    t.equals(result.done, false, 'done');
+
+    const view = result.value;
+    t.equals(view.byteOffset, 0, 'byteOffset');
+    t.equals(view.byteLength, 16, 'byteLength');
+
+    return reader.read();
+  }).then(result => {
+    t.equals(result.done, true, 'done');
+    t.equals(result.value, undefined, 'value');
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: enqueue(), close(), getReader(), then read()', t => {
+  const rbs = new ReadableByteStream({
+    start(c) {
+      c.enqueue(new Uint8Array(16));
+      c.close();
+    },
+    pull() {
+      t.fail('pull must not be called');
+      t.end();
+    },
+    pullInto() {
+      t.fail('pullInto must not be called');
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  reader.read().then(result => {
+    t.equals(result.done, false, 'done');
+
+    const view = result.value;
+    t.equals(view.byteOffset, 0, 'byteOffset');
+    t.equals(view.byteLength, 16, 'byteLength');
+
+    return reader.read();
+  }).then(result => {
+    t.equals(result.done, true, 'done');
+    t.equals(result.value, undefined, 'value');
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: read(), then enqueue()', t => {
+  let controller;
+
+  const rbs = new ReadableByteStream({
+    start(c) {
+      controller = c;
+    },
+    pull() {
+      controller.enqueue(new Uint8Array(16));
+    },
+    pullInto() {
+      t.fail('pullInto must not be called');
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  reader.read().then(result => {
+    t.equals(result.done, false, 'done');
+    t.equals(result.value.byteLength, 16, 'byteLength');
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: read(), then respond()', t => {
+  let controller;
+
+  const rbs = new ReadableByteStream({
+    start(c) {
+      controller = c;
+    },
+    pull() {
+      try {
+        controller.respond(1);
```

Comment:
> This is throwing because it's respond()ing but not to a pullInto(), right? I think the test title should say that, and also you should respond with a Uint8Array or something, because otherwise I'd expect a TypeError anyway when respond()ing with a number.

Sorry. I'll update the test titles to describe what's being tested more clearly.

`pull()` must not responded by `respond()` as there's no existing `pullInto()` i.e. no pending buffer to fill.

https://github.com/whatwg/streams/commit/ffaea95c7ea3acb55fdc371895a7c9476049fc85